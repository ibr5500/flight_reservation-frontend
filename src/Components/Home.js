import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchAirlines } from '../redux/airlines/airlines';
import Airline from './Airline';

function Home() {
  const dataList = useSelector((state) => state.airlines.airlines.data);
  const dispach = useDispatch();

  useEffect(() => {
    dispach(fetchAirlines());
  }, [dispach]);

  return (
    <section className="main">
      <div className="header">
        <h2>Flights list:</h2>
        <p>Please book a flight from the following list</p>
      </div>
      <div className="container">
        <div className="flights">
          {(dataList) ? dataList.map((item) => (
            <Airline
              key={item.id}
              id={item.id}
              image={item.image}
              airlineName={item.name}
              price={item.price}
            />
          )) : null }
        </div>
      </div>
    </section>
  );
}

export default Home;
