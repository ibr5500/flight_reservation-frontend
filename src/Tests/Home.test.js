import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';

import Home from '../Components/Home';

it('renders home page correctly', () => {
  const tree = renderer.create(<Provider store={store}><Home /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
