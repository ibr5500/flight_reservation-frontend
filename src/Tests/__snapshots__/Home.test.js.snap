// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders home page correctly 1`] = `
<section
  className="main"
>
  <div
    className="header"
  >
    <h2>
      Flights list:
    </h2>
    <p>
      Please book a flight from the following list
    </p>
  </div>
  <div
    className="container"
  >
    <div
      className="flights"
    />
  </div>
</section>
`;
